.github/workflows/reusable_gpu.yml

```diff
@@ -15,30 +33,30 @@ env:
   COVERAGE_DIR : "${{github.workspace}}/coverage"
 
 jobs:
-  gpu-Level-Zero:
-    name: Level-Zero
+  gpu:
+    name: "${{inputs.name}}: ${{matrix.os}}, ${{matrix.build_type}}, shared=${{matrix.shared_library}}"
     env:
       VCPKG_PATH: "${{github.workspace}}/../../../../vcpkg/packages/hwloc_x64-windows;${{github.workspace}}/../../../../vcpkg/packages/tbb_x64-windows;${{github.workspace}}/../../../../vcpkg/packages/jemalloc_x64-windows"
-      COVERAGE_NAME : "exports-coverage-gpu"
+      COVERAGE_NAME : "exports-coverage-${{inputs.name}}"
     # run only on upstream; forks will not have the HW
     if: github.repository == 'oneapi-src/unified-memory-framework'
     strategy:
+      fail-fast: false
       matrix:
-        shared_library: ['ON', 'OFF']
-        os: ['Ubuntu', 'Windows']
-        build_type: ['Debug', 'Release']
+        shared_library: ${{ fromJSON(inputs.shared_lib)}}
+        os: ${{ fromJSON(inputs.os)}}
+        build_type: ${{ fromJSON(inputs.build_type)}}
         include:
           - os: 'Ubuntu'
             compiler: {c: gcc, cxx: g++}
             number_of_processors: '$(nproc)'
-          - os: 'Windows'
-            compiler: {c: cl, cxx: cl}
-            number_of_processors: '$Env:NUMBER_OF_PROCESSORS'
-        exclude:
-          - os: 'Windows'
-            build_type: 'Debug'
+          # FIXME:
+          # Temporarily disable including Windows options until we have CUDA enabled.
```

Comment:
I don't know why these options matter to CUDA. You set them as hardcoded anyway...